shell/src/main/java/alluxio/cli/fsadmin/command/BackupCommand.java

```diff
@@ -56,33 +64,64 @@ public String getCommandName() {
 
   @Override
   public Options getOptions() {
-    return new Options().addOption(LOCAL_OPTION);
+    return new Options()
+        .addOption(LOCAL_OPTION)
+        .addOption(ALLOW_LEADER_OPTION);
   }
 
   @Override
   public int run(CommandLine cl) throws IOException {
+    final int progressLineLength = 75;
     String[] args = cl.getArgs();
-    Builder opts = BackupPOptions.newBuilder();
+    BackupPRequest.Builder opts = BackupPRequest.newBuilder();
     if (args.length >= 1) {
       opts.setTargetDirectory(args[0]);
     }
-    opts.setLocalFileSystem(cl.hasOption(LOCAL_OPTION.getLongOpt()));
-    // Create progress thread for showing progress while backup is in progress.
-    Thread progressThread = CommonUtils.createProgressThread(5 * Constants.SECOND_MS, System.out);
-    progressThread.start();
-    BackupResponse resp;
-    try {
-      resp = mMetaClient.backup(opts.build());
-    } finally {
-      progressThread.interrupt();
-    }
-    if (opts.getLocalFileSystem()) {
-      mPrintStream.printf("Successfully backed up journal to %s on master %s with %d entries%n",
-          resp.getBackupUri(), resp.getHostname(), resp.getEntryCount());
-    } else {
-      mPrintStream.printf("Successfully backed up journal to %s with %d entries%n",
-          resp.getBackupUri(), resp.getEntryCount());
-    }
+    opts.setOptions(
+        BackupPOptions.newBuilder()
+            .setRunAsync(true)
+            .setLocalFileSystem(cl.hasOption(LOCAL_OPTION.getLongOpt()))
+            .setAllowLeader(cl.hasOption(ALLOW_LEADER_OPTION.getLongOpt())));
+    // Take backup in async mode.
+    BackupStatus status = mMetaClient.backup(opts.build());
+    do {
+      // Backup could be after a fail-over.
+      if (status.getState() == BackupState.EMPTY) {
+        mPrintStream.printf("Backup lost. Check Alluxio logs.");
+        return -1;
+      }
+      if (status.getState() == BackupState.COMPLETED) {
+        break;
+      } else if (status.getState() == BackupState.FAULTED) {
```

Comment:
Proto doesn't allow it since`meta_master.proto` has another enum that defined `FAILED`. 
I can prefix all values with `Backup_`. Or just leave it as this if not that ugly.

Reply:
So strange, but simple is better, let's maintain FAULTED